### PR TITLE
fix: prevent tool-call preface text leaks

### PR DESCRIPTION
**Pull request by vivy1024**
## Summary
- Suppress assistant preface text on turns that include tool calls so internal tool-use narration is not sent as a user-visible LLM result.
- Normalize OpenAI-compatible text-part repr strings like `[{text=..., type=text}]` to plain text, and drop empty nested repr shells.
- Add regression coverage for both the tool-loop leak and provider content normalization.

## Test plan
- [x] `uv run pytest tests/test_tool_loop_agent_runner.py -k "tool_call_assistant_preface_is_not_sent_as_llm_result"`
- [x] `uv run pytest tests/test_openai_source.py -k "normalize_content_handles_text_part_repr_string or normalize_content_drops_empty_nested_text_part_repr_string"`
- [x] `uv run ruff format astrbot/core/agent/runners/tool_loop_agent_runner.py astrbot/core/provider/sources/openai_source.py tests/test_tool_loop_agent_runner.py tests/test_openai_source.py --check`
- [x] `uv run ruff check astrbot/core/agent/runners/tool_loop_agent_runner.py astrbot/core/provider/sources/openai_source.py tests/test_tool_loop_agent_runner.py tests/test_openai_source.py`

Note: Running the full related files with `uv run pytest tests/test_tool_loop_agent_runner.py tests/test_openai_source.py` passed the tool-loop file but hit 3 pre-existing Windows path separator assertions in `test_openai_source.py` (`file_uri_to_path` expects `/` while Windows returns `\`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Prevent internal assistant preface text from being surfaced as user-visible LLM output during tool-call turns and improve normalization of OpenAI-compatible content representations.

Bug Fixes:
- Suppress LLM reasoning and completion text outputs on steps that invoke tools so assistant preface narration does not leak to users.
- Normalize OpenAI provider content strings that use text-part repr formatting into plain text and remove empty nested text shells.

Tests:
- Add regression tests ensuring tool-call assistant preface text is not returned as an LLM result.
- Add regression tests verifying _normalize_content handles text-part repr strings and drops empty nested text-part repr shells.

### CODE REVIEW

File: tests/test_tool_loop_agent_runner.py [hunk 1]
```diff
@@ -507,6 +507,36 @@ async def snapshot_context_manager(messages, trusted_token_usage=0):
...
+async def test_tool_call_assistant_preface_is_not_sent_as_llm_result(
+    runner, provider_request, mock_tool_executor, mock_hooks, mock_provider
+):
+    """Assistant preface text on a tool-call turn must stay internal."""
+    mock_provider.should_call_tools = True
+    mock_provider.max_calls_before_normal_response = 1
+
+    await runner.reset(
+        provider=mock_provider,
+        request=provider_request,
```
[the remark this answers is not in the document]
Comment (vivy1024) [reply tests/test_tool_loop_agent_runner.py:520]: Addressed in b5bc7b405: the test now asserts exactly one llm_result is emitted, so the tool-call turn cannot surface an assistant preface as user-visible output.

File: tests/test_openai_source.py [hunk 2]
```diff
@@ -60,6 +60,20 @@ def _make_groq_provider(overrides: dict | None = None) -> ProviderGroq:
     )
 
 
+def test_normalize_content_handles_text_part_repr_string():
+    raw = "[{text=I will use the file reading tool., type=text}]"
+
+    assert ProviderOpenAIOfficial._normalize_content(raw) == (
```
[the remark this answers is not in the document]
Comment (vivy1024) [reply tests/test_openai_source.py:66]: Addressed in b5bc7b405: added strip=False coverage for repr-like text-part strings with surrounding whitespace.

File: tests/test_tool_loop_agent_runner.py [hunk 3]
```diff
@@ -507,6 +507,36 @@ async def snapshot_context_manager(messages, trusted_token_usage=0):
...
+        run_context=ContextWrapper(context=None),
+        tool_executor=mock_tool_executor,
+        agent_hooks=mock_hooks,
+        streaming=False,
+    )
+
+    responses = []
+    async for response in runner.step_until_done(3):
+        responses.append(response)
+
```
[the remark this answers is not in the document]
Comment (vivy1024) [reply tests/test_tool_loop_agent_runner.py:530]: Addressed in b5bc7b405: the test now asserts the response type sequence has no llm_result before the final response.

File: tests/test_openai_source.py [hunk 4]
```diff
@@ -60,6 +60,20 @@ def _make_groq_provider(overrides: dict | None = None) -> ProviderGroq:
...
+        "I will use the file reading tool."
+    )
+
+
+def test_normalize_content_drops_empty_nested_text_part_repr_string():
+    raw = "[{text=[{text=[{text=, type=text}], type=text}], type=text}]"
+
+    assert ProviderOpenAIOfficial._normalize_content(raw) == ""
```
[the remark this answers is not in the document]
Comment (vivy1024) [reply tests/test_openai_source.py:82]: Addressed in b5bc7b405: added a nested non-empty repr-like text-part case and asserted it normalizes to the innermost text.

**Comment by Soulter** [comment]
Thanks for the PR. I checked this path, and the intermediate assistant text emitted during a tool-call turn is expected behavior.

Some models can return both `content` and `tool_calls` in the same assistant turn. That text is the model's explanation or transition before calling tools, and it is a valid assistant output. Suppressing it globally in `tool_loop_agent_runner` would change the current tool-loop semantics and may break scenarios that intentionally display intermediate assistant text.

If a specific platform or response stage sends duplicated or poorly placed text, we should handle that at the respond/display layer with deduplication or presentation policy, not by dropping every `llm_result` whenever `tools_call_name` is present.

So I do not recommend merging this direction as-is.

**Comment by Soulter** [comment]
Closing this because the proposed suppression changes expected tool-loop behavior. Intermediate assistant text during tool-call turns is intentional; any duplicate or presentation-specific issue should be handled in the response/display layer instead.
